api/src/pcapi/routes/backoffice/offer_validation_rules/blueprint.py

```diff
@@ -61,6 +61,34 @@ def _get_offerers_data_for_rules(rules: list[offers_models.OfferValidationRule])
     return offerer_dict
 
 
+def _get_venues_data_for_rules(rules: list[offers_models.OfferValidationRule]) -> dict:
+    all_venue_ids: set = set()
+    for rule in rules:
+        for sub_rule in rule.subRules:
+            if (
+                sub_rule.model == offers_models.OfferValidationModel.VENUE
+                and sub_rule.attribute == offers_models.OfferValidationAttribute.ID
+            ):
+                all_venue_ids |= set(sub_rule.comparated["comparated"])
+
+    if not all_venue_ids:
+        return {}
+
+    venues_from_rules = (
+        offerers_models.Venue.query.options(
+            sa.orm.load_only(
+                offerers_models.Venue.id,
+                offerers_models.Venue.name,
```

Comment:
Pour `common_name`, il faut aussi charger `publicName`, sinon une requête N+1 à chaque itération.
Aucun `assert_num_queries` ne t'a permis de le détecter ?

Reply:
Il semble que non, mais les tests sont avec des règles sur un seul lieu, ce qui cache le problème